// _/test/test.js
// Generated by CoffeeScript 1.6.3
var _;

_ = require('../_');

exports['each (object)'] = function(test) {
  var accumulator, expected, items;
  items = {
    one: 'foo',
    two: 'bar',
    tre: 'baz'
  };
  accumulator = '';
  expected = 'foobarbaz';
  _.each(items, function(value) {
    return accumulator += value;
  });
  test.equal(accumulator, expected);
  return test.done();
};

exports['each (array)'] = function(test) {
  var accumulator, expected, items;
  items = [1, 1, 2, 3, 5, 8, 13, 21];
  accumulator = 0;
  expected = 54;
  _.each(items, function(value) {
    return accumulator += value;
  });
  test.equal(accumulator, expected);
  return test.done();
};

exports.extend = function(test) {
  var four, one, three, two;
  one = {
    foo: 1
  };
  two = {
    bar: 2
  };
  three = {
    baz: 3
  };
  four = {
    baz: 4
  };
  _.extend(one, two, three, four);
  test.equal(one.foo, 1);
  test.equal(one.bar, 2);
  test.equal(one.baz, 4);
  return test.done();
};

exports.fluent = function(test) {
  var obj;
  obj = {
    fn: _.fluent(function() {
      return 'foo';
    })
  };
  test.deepEqual(obj.fn(), obj);
  return test.done();
};

exports.one = function(test) {
  var items;
  items = {
    one: 'foo',
    two: 'bar',
    tre: 'baz'
  };
  test.equal(_.one(items), 'one');
  return test.done();
};
